File: csharp/log-levels/LogLevels.cs
using System;

static class LogLine
{
    public static string Message(string logLine)
    {
        int endOfMessageLevelIdentifier = logLine.IndexOf(":") + 1;

        return logLine.Substring(endOfMessageLevelIdentifier).Trim();
    }

    public static string LogLevel(string logLine) => logLine.Replace(Message(logLine), string.Empty)
                                                            .Replace("[", string.Empty)
                                                            .Replace("]: ", string.Empty)
                                                            .TrimEnd()
                                                            .ToLower();

    public static string Reformat(string logLine) => $"{Message(logLine)} ({LogLevel(logLine)})";
}

File: fsharp/log-levels/LogLevels.fs
module LogLevels

let message (logLine: string) = logLine.Split(':').[1].Trim()

let logLevel(logLine: string) = logLine.Split(':').[0].Replace("[","").Replace("]","").Trim().ToLower()

let reformat(logLine: string) = message logLine + " (" + logLevel logLine + ")" 
